### PR TITLE
phpExtensions: prevent `buildPecl` to always use the same version of PHP

### PR DESCRIPTION
**Pull request by drupol**
This PR:

Is a follow up of #242366 and #242166 

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb

File: pkgs/top-level/php-packages.nix [hunk 1]
```diff
@@ -247,9 +247,9 @@ lib.makeScope pkgs.newScope (self: with self; {
 
     memcached = callPackage ../development/php-packages/memcached { };
 
-    mongodb = pkgs.darwin.apple_sdk_11_0.callPackage ../development/php-packages/mongodb {
-      inherit (pkgs.darwin.apple_sdk_11_0.frameworks) Security;
-      inherit (pkgs.darwin.apple_sdk_11_0) Libsystem;
+    mongodb = callPackage ../development/php-packages/mongodb {
+      inherit (pkgs.darwin.apple_sdk_11_0) stdenv;
```
Comment (wegank) [inline pkgs/top-level/php-packages.nix:251]: Does it work on Linux?

Comment (drupol) [reply pkgs/top-level/php-packages.nix:251]: yep looks like, it's building here.

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb

File: pkgs/development/php-packages/mongodb/default.nix [hunk 2]
```diff
@@ -35,8 +34,8 @@ php.buildPecl rec {
     zlib
     pcre2
   ] ++ lib.optionals stdenv.isDarwin [
-    Security
-    Libsystem
+    darwin.apple_sdk_11_0.frameworks.Security
+    darwin.Libsystem
```
[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/development/php-packages/mongodb/default.nix]: Fixed.

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb php81Extensions.datadog_trace

File: pkgs/top-level/php-packages.nix (hunk 1 above)
Comment (wegank) [reply pkgs/top-level/php-packages.nix:251]: I'm sure `darwin.apple_sdk_11_0.stdenv` doesn't evaluate on Linux. So `stdenv` seems pretty unused here.

Comment (drupol) [reply pkgs/top-level/php-packages.nix:251]: You're right, I added 
image

and

image

And now I get an error trying to evaluate it.

I guess I can get rid of it?

Comment (wegank) [reply pkgs/top-level/php-packages.nix:251]: I think `stdenv = if stdenv.isDarwin then pkgs.darwin.apple_sdk_11_0.stdenv else stdenv` is the most common pattern. A `// lib.optionalAttrs stdenv.isDarwin { inherit (pkgs.darwin.apple_sdk_11_0) stdenv; }` should also do the trick, though.

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb php81Extensions.datadog_trace

File: pkgs/top-level/php-packages.nix (hunk 1 above)
Comment (drupol) [reply pkgs/top-level/php-packages.nix]: I removed it for now, it seems to build properly without it... let's wait and see the CI.

**Comment by 2xsaiko** [comment]
> php81Extensions.mongodb php81Extensions.datadog_trace

These build successfully on my aarch64-darwin machine. (couldn't do nixpkgs-review because of #242680)

**Pull request merged**